src/NLog/Config/ConfigurationItemFactory.cs

```diff
@@ -223,13 +223,70 @@ public void RegisterItemsFromAssembly(Assembly assembly, string itemNamePrefix)
         {
             InternalLogger.Debug("ScanAssembly('{0}')", assembly.FullName);
             var typesToScan = assembly.SafeGetTypes();
+            PreloadAssembly(typesToScan);
             foreach (IFactory f in this.allFactories)
             {
                 f.ScanTypes(typesToScan, itemNamePrefix);
             }
         }
 
         /// <summary>
+        /// Call Preload for NLogPackageLoader
+        /// </summary>
+        /// <param name="typesToScan"></param>
+        public void PreloadAssembly(Type[] typesToScan)
+        {
+            var types = typesToScan.Where(t => t.Name.Equals("NLogPackageLoader", StringComparison.OrdinalIgnoreCase));
+
+            foreach (var type in types)
+            {
+                CallPreload(type);
+            }
+        }
+
+        /// <summary>
+        /// Call Preload for <paramref name="type"/>. The method should be static.
+        /// </summary>
+        /// <param name="type"></param>
+        private static void CallPreload(Type type)
+        {
+            if (type != null)
+            {
+                InternalLogger.Debug("Found for preload'{0}'", type.FullName);
+                var preloadMethod = type.GetMethod("Preload");
+                if (preloadMethod != null)
+                {
+                    if (preloadMethod.IsStatic)
+                    {
+
+                        InternalLogger.Debug("NLogPackageLoader contains preload method");
+                        //only static, so first param null
+                        try
+                        {
+                            preloadMethod.Invoke(null, null);
+                        }
+                        catch (Exception e)
+                        {
+                            InternalLogger.Warn(e,"Preload invoked for '{0}' failed", type.FullName);
+                        }
+                       
+                        InternalLogger.Debug("Preload succesfully invoked for '{0}'", type.FullName);
```

Comment:
This line will be executed, even if invoke has been failed. I suggest that move this line to the line 267 (after the invoke).